helixbass/tree-sitter-grep
# WIP: show heading + color

In this PR:
- mimic `ripgrep`'s heading/filename/line/column formatting (at least much more closely), eg showing a per-file "heading" and colors by default when outputting to a terminal

To test:
Now when running in a terminal by default you should see `ripgrep`-style output formatting (but with a different default set of colors)
If you eg redirect the output by doing `| cat -`, you should see different formatting that is more like our existing formatting but without showing line numbers (by default)
You should be able to use the newly exposed command-line arguments to modify those behaviors, eg use `--pretty` when redirecting the output to still get color/headings etc, use `--line-number` when redirecting the output to still get line numbers
